Alien.cpp

```diff
@@ -1,11 +1,24 @@
 #include "Alien.h"
 #include <iostream>
 
+foreign_life::Alien::Alien(std::string a_name) :
+    name{std::move(a_name)}
+{}
+
+foreign_life::Alien::~Alien() {}
+
 void foreign_life::Alien::set_type(int type) {
     this->type = type;
 }
 
-void foreign_life::Alien::print_type() const {
-    std::cout << "Type of alien is " << this->type
-         << std::endl;
+int foreign_life::Alien::get_type() const {
+    return this->type;
```

Comment:
I think this style of explicitly using `this->` is highly unusual.
The vast majority of C++ coders would just write `type` not `this->type`.

Reply:
Is this a new feature of modern C++? I just remember in school always being taught to use this pointer